examples/Playground.hs

```diff
@@ -58,35 +57,17 @@ main = do
         Right test -> Toml.encode testT test
 
 myToml :: TOML
-myToml = TOML (HashMap.fromList
-    [ ("a"   , AnyValue $ Bool True)
-    , ("list", AnyValue $ Array [String "one", String "two"])
-    , ("time", AnyValue $ Array [Date $ Day (fromGregorian 2018 3 29)])
-    ] ) myInnerToml
-
-myInnerToml :: PrefixMap TOML
-myInnerToml = fromList
-    [ ( "table.name.1"
-      , TOML (HashMap.fromList
-            [ ("aInner"   , AnyValue $ Int 1)
-            , ("listInner", AnyValue $ Array [Bool True, Bool False])
-            ]) myInnerInnerToml
-      )
-    , ( "table.name.2"
-      , TOML (HashMap.fromList [("2Inner", AnyValue $ Int 42)]) mempty
-      )
-    ]
-
-
-myInnerInnerToml :: PrefixMap TOML
-myInnerInnerToml = fromList
-    [ ( "table.name.1.1"
-      , TOML (HashMap.fromList
-            [ ("aInner"   , AnyValue $ Int 1)
-            , ("listInner", AnyValue $ Array [Bool True, Bool False])
-            ]) mempty
-      )
-    , ( "table.name.1.2"
-      , TOML (HashMap.fromList [("Inner1.2", AnyValue $ Int 42)]) mempty
-      )
-    ]
+myToml = toml $ do
+    "a" =: Bool True
+    "list" =: Array [String "one", String "two"]
+    "time" =: Array [Date $ Day (fromGregorian 2018 3 29)]
+    table "table.name.1" $ do
+        "aInner" =: Int 1
```

Comment:
I think I know how you can write `key =: 1` instead of `ke =: Int 1`. Since we have GADT for `Value`, you can implement `Num` instance for `Value 'TInt`. And this will guarantee that it will be converted to `Value` using `Int` constructor. Also all pattern matchings will be exhaustive inside instance implementation because of that!

Same for `IsString` instance for `Value`. Not sure about `IsList` intsance, it probably will be impossible to implement... And, unfortunately, there's no `IsBool` type class in Haskell :disappointed: